aws/resources/transit_gateway.go

```diff
@@ -60,6 +70,122 @@ func (tgw *TransitGateways) getAll(c context.Context, configObj config.Config) (
 	return ids, nil
 }
 
+func (tgw *TransitGateways) nuke(id *string) error {
+	// check the transit gateway has attachments and nuke them before
+	if err := tgw.nukeAttachments(id); err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	if _, err := tgw.Client.DeleteTransitGatewayWithContext(tgw.Context, &ec2.DeleteTransitGatewayInput{
+		TransitGatewayId: id,
+	}); err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	return nil
+}
+
+func (tgw *TransitGateways) nukeAttachments(id *string) error {
+	logging.Debugf("nuking transit gateway attachments for %v", awsgo.StringValue(id))
+	output, err := tgw.Client.DescribeTransitGatewayAttachmentsWithContext(tgw.Context,&ec2.DescribeTransitGatewayAttachmentsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: awsgo.String("transit-gateway-id"),
+				Values: []*string{
+					id,
+				},
+			},
+			{
+				Name: awsgo.String("state"),
+				Values: []*string{
+					awsgo.String("available"),
+				},
+			},
+		},
+	})
+	if err != nil {
+		logging.Errorf("[Failed] unable to describe the  transit gateway attachments for %v : %s", awsgo.StringValue(id), err)
+		return errors.WithStackTrace(err)
+	}
+
+	logging.Debugf("%v attachment(s) found with %v", len(output.TransitGatewayAttachments),awsgo.StringValue(id))
+
+	for _, attachments := range output.TransitGatewayAttachments {
+		var  (
+			err error
+			attachmentType = awsgo.StringValue(attachments.ResourceType)
+			now = time.Now()
+		)
+
+		switch attachmentType {
+		case TransitGatewayAttachmentTypePeering:
+			logging.Debugf("[Execution] deleting the attachments of type %v for %v ", attachmentType, awsgo.StringValue(id))
+			_, err = tgw.Client.DeleteTransitGatewayPeeringAttachmentWithContext(context.Background(), &ec2.DeleteTransitGatewayPeeringAttachmentInput{
+				TransitGatewayAttachmentId: attachments.TransitGatewayAttachmentId,
+			})
+		default:
+			err = fmt.Errorf("%v typed transit gateway attachment nuking not handled.", attachmentType)
```

Comment:
lint suggestion: Error string should not be capitalized or end with punctuation mark